src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs

```diff
@@ -569,26 +568,41 @@ private static void VerifyRefAssemblyClient(string lib_cs, string source, Action
 
         [Theory]
         [InlineData("public int M() { error(); }", true)]
-        [InlineData("public int M() { error() }", false)] // Should be true. See follow-up issue https://github.com/dotnet/roslyn/issues/17612
+        [InlineData("public int M() { error() }", false)] // This may get relaxed. See follow-up issue https://github.com/dotnet/roslyn/issues/17612
+        [InlineData("public int M();", true)]
+        [InlineData("public int M() { int Local(); }", true)]
+        [InlineData("public C();", true)]
+        [InlineData("~ C();", true)]
         [InlineData("public Error M() { return null; }", false)] // This may get relaxed. See follow-up issue https://github.com/dotnet/roslyn/issues/17612
+        [InlineData("public static explicit operator C(int i);", true)]
+        [InlineData("public async Task M();", false)]
+        [InlineData("partial void M(); partial void M();", false)] // This may get relaxed. See follow-up issue https://github.com/dotnet/roslyn/issues/17612
```

Comment:
@gafter I just added this test for "partial". Emitting this with metadata-only still gives an error (`error CS0756: A partial method may not have multiple defining declarations`).
I plan to leave this as-is for C# 7.1, as this does not seem a requirement for the corefx scenario.